rodi/__init__.py

```diff
@@ -562,10 +562,8 @@ def _resolve_by_annotations(
         fns = self._get_resolvers_for_parameters(concrete_type, context, params)
         resolvers = {}
 
-        i = 0
-        for name in annotations.keys():
+        for i, name in enumerate(annotations.keys()):

```

Comment:
Probably, `index` variable name would be more meaningful.

Reply:
I think `i` in this context is clear.